### DISCUSSION TITLE
10X sc TCR sequencing outputs

### DISCUSSION THREAD

**Discussion by bshim181** [milaboratory/mixcr#1400] [Q&A] [answered]
Hello, 

I was comparing the outputs from MiXCR and Cellranger and I noticed that there are some differences between number of cell barcodes identified and the chains it gets assigned with. I was first wondering what annotation MiXCR uses as default? is it based off of IMGT? or is it more of custom generated? 

Furthermore, I noticed from two outputs, Cellranger did not have any cell barcodes that were supported under 25 reads and I was wondering if I could enforce some kind of threshold for contig/barcode detection with read number. 

Lastly, I noticed that there were multi chain grouping functionality added to the MiXCR in Version 4.5.0. Is it possible to explain how the multi-chain clonotypes have been determined (brief explanation of the algorithm behind it)?

**Comment by ejohnson643** [comment]
Bumping this! I would also really like to know how any multi-chain clonotypes are identified!

**Comment by mizraelson** [comment, marked as answer]
Hi,

 - By default, MiXCR utilizes its own built-in library. However, if required, you can [use IMGT](https://mixcr.com/mixcr/guides/external-libraries/).

- MiXCR automatically calculates dynamic thresholds for Reads per UMI, UMIs per cell, and several other advanced filters. These filters are determined based on the barcode distribution in each sample to optimize performance when applied collectively. Filtering all cells by a fixed number of reads (e.g., 25) wouldn't be suitable in many scenarios. While it's possible to write a new preset to hardcode a 25 reads/cell limit, I'd strongly advise against it. Based on our experience, the filters we employ provide more accurate immune repertoire analysis.

- The grouping algorithm operates in the following manner: We first filter out non-productive clones and cell barcodes that combine chains, which are obviously from duplets (e.g., a combination of TCR and BCR or IGK and IGL in a single cell). Clones are then grouped by CELL barcodes. Subsequently, we attempt to merge cell groups, adding to the most substantial group any group that is fully encompassed by the target. This method results in the formation of cell groups sharing the same chains. We plan on refining this algorithm in the next release to better cater to data with significant noise.

Sincerely,
Mark

**Reply by PoslavskySV** [reply]
In addition, two useful links:
 - https://github.com/milaboratory/mixcr/discussions/1294
 - https://github.com/milaboratory/mixcr/discussions/1311

**Reply by ejohnson643** [reply]
Thank you so much! This is very useful.

**Reply by bshim181** [reply]
Thank you!
